Use logical operators for boolean operands

hahnjo opened the pull request:
This addresses a warning of recent Clang versions about the `use of bitwise '|' with boolean operands [-Wbitwise-instead-of-logical]`.

hahnjo commented:
@phsft-bot build on ROOT-performance-centos8-multicore/cxx17, windows10/cxx14

hahnjo commented:
ping... (in my opinion, the test failures aren't related to the changes)

hahnjo commented:
another ping @Axel-Naumann @guitargeek @lmoneta ...

hahnjo commented:
a third ping @Axel-Naumann @guitargeek @lmoneta 
https://github.com/root-project/root/pull/10832 introduced two more instances of this warning, which I've now added to this PR.

Axel-Naumann commented on interpreter/cling/lib/Utils/Output.cpp:
```diff
@@ -80,10 +80,12 @@ namespace cling {
+      bool colorStdout = COLOR_FLAG(1, StandardOutIsDisplayed);
+      bool colorStderr = COLOR_FLAG(2, StandardErrIsDisplayed);
+      // The following calls have side effects because they set m_Colorize!
+      static_cast<ColoredOutput &>(outs()).Colors(colorStdout);
+      static_cast<ColoredOutput &>(errs()).Colors(colorStderr);
```
Before, the second call might have been shortcut, which makes this a functional change because now `m_Colorize` might be touched for `StandardErrIsDisplayed` where it wasn't touched before. Is that an improvement / bug fix or an unintentional side-effect or not actually happening? Can you call this out in the commit log?

hahnjo replied:
No, `|` is not short-circuited because it's a bitwise operation where both operands have to be known. When I replaced with the logical `||`, it was short-circuited and tests started to fail. So actually, this is preserving the status-quo and I added a code comment to prevent somebody clever to come around and "optimize" this in the future.

Axel-Naumann reviewed (approved):
LGTM but see comment on change in cling. I would confidently vouch for the RooFit changes to be just big fixes. So please go ahead and merge even without @lmoneta 's review, possibly after addressing the short-circuit commit log.

hahnjo commented:
> #10832 introduced two more instances of this warning, which I've now added to this PR.

Removed again as part of #10966, force-pushing without touching these files and then merging (unless Jenkins complains)
